lib/double_entry/transfer.rb

```diff
@@ -99,22 +99,27 @@ def create_lines(amount, code, detail, from_account, to_account, metadata)
       credit_balance = Locking.balance_for_locked_account(from_account)
       debit_balance  = Locking.balance_for_locked_account(to_account)
 
-      credit_balance.update_attribute :balance, credit_balance.balance - amount
-      debit_balance.update_attribute :balance, debit_balance.balance + amount
+      credit_balance_update_amount = credit_balance.balance - amount
+      debit_balance_update_amount = debit_balance.balance + amount
 
       credit.amount, debit.amount   = -amount, amount
       credit.account, debit.account = from_account, to_account
       credit.code, debit.code       = code, code
       credit.detail, debit.detail   = detail, detail
-      credit.balance, debit.balance = credit_balance.balance, debit_balance.balance
+      credit.balance, debit.balance = credit_balance_update_amount, debit_balance_update_amount
       credit.metadata, debit.metadata = metadata, metadata if DoubleEntry.config.json_metadata
 
       credit.partner_account, debit.partner_account = to_account, from_account
 
       credit.save!
-      debit.partner_id = credit.id
       debit.save!
+
       credit.update_attribute :partner_id, debit.id
+      debit.update_attribute :partner_id, credit.id
```

Comment:
I think this one should be
`debit.update_attributes! partner_id: credit.id`

Reply:
on previous process `save!` will raise error
so I think this one should raise too
to make sure data is consistent

Reply:
I had an issue with these code not saving the ids. Even though it was in a transaction, I have no idea why. 

For me, the answer would be to switch to UUID ids, and set them all in one create statement.

Reply:
what is the error message when you were failing to save it?

Reply:
No error, just the ids were blank. Yes I know, crazy. I was using the que gem, which uses DB callbacks/stored proceedures to do background jobs, may be related. But a transaction UUID would allow:

- one save, rather than updates
- in the future, multiple transactions as part of double entry. ( Rather than partner_id, have transaction ID ). 

The latter feature is the following. Just because it's called "Double Entry" doesn't mean there should be only lines in a transaction. 

Can be 5+5 = 10. I.e. ( Cost of Good, Profit = Sales )

I've now used double_entry gem in 3 *production* systems!

Reply:
yes agreed about this
`Can be 5+5 = 10. I.e. ( Cost of Good, Profit = Sales )`
it's a feature very good to have

in your production systems, are you using this commit into your double_entry gem?